lib/langchain/llm/responses/google_palm.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Langchain::LLM::Responses::GooglePalm
```

Comment:
Why not just do a classic class inheritance structure, like the following (oversimplified) below?

```ruby
class Langchain::LLM::Response
  def initialize(raw_response)
     @raw_response = raw_response
  end

  def completion_text
```

```ruby
class Langchain::LLM::GooglePalmResponse < Langchain::LLM::Response
  def completion_text
     completions.first.dig("output")
  end

  def completions
    raw_response.dig("candidates")
  end
```

Reply:
Well, that's one way to do it. In this case, all the LLMs will return instances of different classes. I was trying to achieve a consistent result.